File: src/main/java/client/inventory/operations/AddInventoryOperation.java
package client.inventory.operations;

import client.inventory.ItemInventoryType;
import client.inventory.ModifyInventoryOperationType;
import client.inventory.slots.ItemSlot;
import net.maple.packets.ItemPackets;
import util.packet.PacketWriter;

public class AddInventoryOperation extends AbstractModifyInventoryOperation {

    private final ItemSlot item;

    public AddInventoryOperation(ItemInventoryType inventory, short slot, ItemSlot item) {
        super(inventory, slot);
        this.item = item;
    }

    @Override
    protected ModifyInventoryOperationType getType() {
        return ModifyInventoryOperationType.ADD;
    }

    @Override
    protected void encodeData(PacketWriter pw) {
        ItemPackets.encode(item, pw);
    }
}

File: src/main/java/client/inventory/operations/AddInventoryOperation.kt
package client.inventory.operations

import client.inventory.ItemInventoryType
import client.inventory.ModifyInventoryOperationType
import client.inventory.item.slots.ItemSlot
import net.maple.packets.ItemPackets.encode
import util.packet.PacketWriter

class AddInventoryOperation(inventory: ItemInventoryType, slot: Short, private val item: ItemSlot) : AbstractModifyInventoryOperation(inventory, slot) {

    override val type = ModifyInventoryOperationType.ADD

    override fun encodeData(pw: PacketWriter) {
        item.encode(pw)
    }
}